src/Adapter/Driver/IbmDb2/Statement.php

```diff
@@ -172,7 +174,14 @@ public function prepare($sql = null)
             $sql = $this->sql;
         }
 
-        $this->resource = db2_prepare($this->db2, $sql);
+        try {
+            set_error_handler($this->createErrorHandler());
+            $this->resource = db2_prepare($this->db2, $sql);
+        } catch (Throwable $e) {
```

Comment:
should catch `Exception` for php 5.5/5.6 fallback as well ?

Reply:
No - in 5.6, nothing is thrown.